File: backend/src/Contracts/Patients/PatientContactUpdateRequest.cs
namespace OptimaHealthcare.Contracts.Patients;

/// <summary>
/// Contact details only - for patient self-update.
/// </summary>
public sealed class PatientContactUpdateRequest
{
    public long MobileNo { get; set; }
    public string Email { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
}

File: mobile/lib/features/patients/models/patient_contact_update_request.dart
class PatientContactUpdateRequestModel {
  PatientContactUpdateRequestModel({
    required this.mobileNo,
    required this.email,
    required this.address,
    required this.city,
  });

  final int mobileNo;
  final String email;
  final String address;
  final String city;

  Map<String, dynamic> toJson() {
    return {
      'mobileNo': mobileNo,
      'email': email,
      'address': address,
      'city': city,
    };
  }
}
